lib/shipit/environment_variables.rb

```diff
@@ -0,0 +1,36 @@
+module Shipit
+  class EnvironmentVariables
+    NotPermitted = Class.new(StandardError)
+
+    class << self
+      def with(env)
+        EnvironmentVariables.new(env)
+      end
+    end
+
+    def permit(variable_definitions)
+      return {} unless @env
+      raise "A whitelist is required to sanitize environment variables" unless variable_definitions
+      sanitize_env_vars(variable_definitions)
+    end
+
+    private
+
+    def initialize(env)
+      @env = env
+    end
+
+    def sanitize_env_vars(variable_definitions)
+      allowed_variables = variable_definitions.map(&:name)
+
+      all_envs = @env.group_by { |k, _| allowed_variables.include?(k) }
+      allowed = all_envs[true].to_h
+      disallowed = all_envs[false].to_h
+
+      error_message = "Variables #{disallowed.keys} have not been whitelisted"
```

Comment:
Might want to use `to_sentence` here: http://api.rubyonrails.org/classes/Array.html#method-i-to_sentence